mmdet3d/evaluation/functional/panoptic_seg_eval.py

```diff
@@ -0,0 +1,384 @@
+            output_dict[class_str]['rq'] = _rq
+            output_dict[class_str]['miou'] = _iou
+
+        pq_dagger = np.mean(
+            [float(output_dict[c]['pq']) for c in self.things_classes] +
+            [float(output_dict[c]['miou']) for c in self.stuff_classes])
+
+        pq_things = np.mean(
+            [float(output_dict[c]['pq']) for c in self.things_classes])
+        rq_things = np.mean(
+            [float(output_dict[c]['rq']) for c in self.things_classes])
+        sq_things = np.mean(
+            [float(output_dict[c]['sq']) for c in self.things_classes])
+
+        pq_stuff = np.mean(
+            [float(output_dict[c]['pq']) for c in self.stuff_classes])
+        rq_stuff = np.mean(
+            [float(output_dict[c]['rq']) for c in self.stuff_classes])
+        sq_stuff = np.mean(
+            [float(output_dict[c]['sq']) for c in self.stuff_classes])
+
+        result_dicts = {}
+        result_dicts['pq'] = float(pq)
+        result_dicts['pq_dagger'] = float(pq_dagger)
+        result_dicts['sq_mean'] = float(sq)
+        result_dicts['rq_mean'] = float(rq)
+        result_dicts['miou'] = float(miou)
+        result_dicts['pq_stuff'] = float(pq_stuff)
+        result_dicts['rq_stuff'] = float(rq_stuff)
+        result_dicts['sq_stuff'] = float(sq_stuff)
+        result_dicts['pq_things'] = float(pq_things)
+        result_dicts['rq_things'] = float(rq_things)
+        result_dicts['sq_things'] = float(sq_things)
+
+        if self.logger is not None:
+            print_log('|        |   IoU   |   PQ   |   RQ   |  SQ   |',
+                      self.logger)
+            for k, v in output_dict.items():
+                print_log(
+                    '|{}| {:.4f} | {:.4f} | {:.4f} | {:.4f} |'.format(
+                        k.ljust(8)[-8:], v['miou'], v['pq'], v['rq'], v['sq']),
+                    self.logger)
+            print_log('True Positive: ', self.logger)
+            print_log('\t|\t'.join([str(x) for x in self.pan_tp]), self.logger)
+            print_log('False Positive: ')
+            print_log('\t|\t'.join([str(x) for x in self.pan_fp]), self.logger)
+            print_log('False Negative: ')
+            print_log('\t|\t'.join([str(x) for x in self.pan_fn]), self.logger)
+
+        else:
+            print('|        |   IoU   |   PQ   |   RQ   |  SQ   |')
+            for k, v in output_dict.items():
+                print('|{}| {:.4f} | {:.4f} | {:.4f} | {:.4f} |'.format(
+                    k.ljust(8)[-8:], v['miou'], v['pq'], v['rq'], v['sq']))
+            print('True Positive: ')
+            print('\t|\t'.join([str(x) for x in self.pan_tp]))
+            print('False Positive: ')
+            print('\t|\t'.join([str(x) for x in self.pan_fp]))
+            print('False Negative: ')
+            print('\t|\t'.join([str(x) for x in self.pan_fn]))
+
+        return result_dicts
+
+    def get_pq(
+        self
+    ) -> Tuple[np.double, np.double, np.ndarray, np.ndarray, np.ndarray]:
```

Comment:
It seems ugly.
```suggestion
    ) -> PQReturnsType:
```
Also, define ` PQReturnsType =  Tuple[np.double, np.double, np.ndarray, np.ndarray, np.ndarray]` in  front of this class.